plugins/GSdx/GSTextureCache.cpp

```diff
@@ -268,25 +270,16 @@ GSTextureCache::Source* GSTextureCache::LookupSource(const GIFRegTEX0& TEX0, con
 				uint32 t_psm = (t->m_dirty_alpha) ? t->m_TEX0.PSM & ~0x1 : t->m_TEX0.PSM;
 
 				if (GSUtil::HasSharedBits(bp, psm, t->m_TEX0.TBP0, t_psm)) {
-					if (!s_IS_OPENGL && (psm == PSM_PSMT8)) {
-						// OpenGL can convert the texture directly in the GPU. Not sure we want to keep this
-						// code for DX. It fixes effect but it is slow (MGS3)
-
-						// It is a complex to convert the code in shader. As a reference, let's do it on the CPU, it will
-						// be slow but
-						// 1/ it just works :)
-						// 2/ even with upscaling
-						// 3/ for both DX and OpenGL
-
-						// Gregory: to avoid a massive slow down for nothing, let's only enable
-						// this code when CRC is below the FULL level
-						if (m_crc_hack_level < CRCHackLevel::Full)
-							Read(t, t->m_valid);
-						else
-							dst = t;
-					} else {
+					// It is a complex to convert the code in shader. As a reference, let's do it on the CPU, it will be slow but
+					// 1/ it just works :)
+					// 2/ even with upscaling
+					// 3/ for both DX and OpenGL
+					if (m_cpu_8bit_fb_coversion && psm == PSM_PSMT8)
+						// Direct3D doesn't have a shader conversion so it needs to be done on the CPU side which causes slowdowns (Fifa Street).
+						// OpenGL has a shader but it doesn't handle some corners properly (Harry Potter, COD ?, Kung fu Panda ?).
```

Comment:
corner cases

Reply:
I updated the comment to match what's written on the tooltip. I think it's better.